source/common/http/conn_pool_grid.cc

```diff
@@ -334,5 +333,50 @@ void ConnectivityGrid::onDrainReceived() {
   }
 }
 
+bool ConnectivityGrid::shouldAttemptHttp3() {
+  if (http3_status_tracker_.isHttp3Broken()) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", host_->hostname());
+    return false;
+  }
+  if (!alternate_protocols_.has_value()) {
+    // TODO(RyanTheOptimist): Make the alternate protocols cache required.
+    ENVOY_LOG(trace, "No alternate protocols cache. Attempting HTTP/3 to host '{}'.",
+              host_->hostname());
+    return true;
+  }
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ASSERT(false);
+    return false;
```

Comment:
nit: add an error log?

Reply:
Done. Is that the common pattern for ASSERTS in Envoy?

Reply:
I don't know what the common pattern is, I was just thinking that most users are on release builds and they'd probably want to know what went wrong.